src/olympia/shelves/forms.py

```diff
@@ -19,27 +19,29 @@ def clean(self):
         criteria = data['criteria']
         baseUrl = settings.INTERNAL_SITE_URL
 
-        if data['shelf_type'] in ('extension', 'search', 'theme'):
+        if data['shelf_type'] in ('extension', 'recommended', 'search',
+                                  'theme'):
             api = drf_reverse('v4:addon-search')
+            url = baseUrl + api + criteria
         elif data['shelf_type'] == 'categories':
             api = drf_reverse('v4:category-list')
+            url = baseUrl + api + criteria
         elif data['shelf_type'] == 'collections':
-            api = drf_reverse('v4:collection-list')
-        elif data['shelf_type'] == 'recommendations':
-            api = drf_reverse('v4:addon-recommendations')
-
-        url = baseUrl + api + criteria
+            api = drf_reverse('v4:collection-addon-list', kwargs={
+                'user_pk': settings.TASK_USER_ID,
+                'collection_slug': criteria
+            })
+            url = baseUrl + api
 
         try:
             response = requests.get(url)
-            results = response.json()
-
             if response.status_code == 404:
                 raise forms.ValidationError('Check criteria - No data found')
             if response.status_code != 200:
                 raise forms.ValidationError(
-                    'Check criteria - %s' % results[0])
+                    'Check criteria - %s' % response.json()[0])
 
+            results = response.json()
```

Comment:
not sure what this refactoring is for.

Reply:
To help retrieve the number of add-ons returned. It's kind of redundant now that `categories` uses the `search` endpoint.